src/App.tsx

```diff
@@ -1,27 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const onClickHandle = async (
+    f: () => Promise<Good[]>,
+  ) => {
+    setGoods(await f());
+  };
```

Comment:
Create 3 separate handlers for buttons

```suggestion
  const handleClickAllGoods = (event: ...) => {
    event.preventDefault();
    setGoods(getAll());
  };
```